lenskit-eval/src/main/java/org/lenskit/eval/traintest/predict/PredictEvalTask.java

```diff
@@ -298,7 +298,15 @@ public PredictConditionEvaluator(TableWriter tw, RatingPredictor pred, List<Metr
         public Map<String, Object> finish() {
             Map<String,Object> results = new HashMap<>();
             for (MetricContext<?> mc: predictMetricContexts) {
-                results.putAll(mc.getAggregateMeasurements().getValues());
+                MetricResult measurements = mc.getAggregateMeasurements();
+                // if test and train tests are disjoint we can get "null"
+                if (measurements != null) {
+                    results.putAll(measurements.getValues());
+                } else {
```

Comment:
I don't think we need the `else` block - if we don't have it, then the missing values will be `null`, which is perfectly acceptable (they'll be emitted as empty cells in the final CSV file). Empty cells work better than NaN, because R will pick them up as NA (which is distinct from NaN in R's world).